### PR TITLE
[backend] Fix of creation with associated file in entities

### LINKED ISSUE #6075: Creating a Report with an associated file gives an error
## Description

If you create a report (also confirmed with  malware analysis) and associate a file in the creation form, we get a backend error.

The object seems created successfully and file is present in data tab, so I'm unsure of the consequences of this error.

## Environment

Testing latest 5.12

NOT PRESENT ON PROD

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. create report with an associated file

## Expected Output

no error

## Actual Output

error message `#<Promise> could not be cloned.`

## Additional information

log error:

```
{
  "category": "APP",
  "errors": [
    {
      "attributes": {
        "genre": "TECHNICAL",
        "http_status": 500
      },
      "message": "#<Promise> could not be cloned.",
      "name": "UNKNOWN_ERROR",
      "stack": "UNKNOWN_ERROR: #<Promise> could not be cloned.\n    at error (/opt/opencti/build/src/config/errors.js:8:10)\n    at UnknownError (/opt/opencti/build/src/config/errors.js:76:47)\n    at Object._logWithError (/opt/opencti/build/src/config/conf.js:331:23)\n    at Object.error (/opt/opencti/build/src/config/conf.js:341:48)\n    at Object.willSendResponse (/opt/opencti/build/src/graphql/loggerPlugin.js:115:20)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async Promise.all (index 1)\n    at S (/opt/opencti/build/node_modules/apollo-server-core/src/requestPipeline.ts:530:5)\n    at processHTTPRequest (/opt/opencti/build/node_modules/apollo-server-core/src/runHttpQuery.ts:437:24)"
    },
    {
      "message": "#<Promise> could not be cloned.",
      "name": "DataCloneError",
      "stack": "DataCloneError: #<Promise> could not be cloned.\n    at new DOMException (node:internal/per_context/domexception:53:5)\n    at createEntityRaw (/opt/opencti/build/src/database/middleware.js:3046:17)\n    at createEntity (/opt/opencti/build/src/database/middleware.js:3183:22)\n    at addReport (/opt/opencti/build/src/domain/report.js:122:25)\n    at Object.resolve [as reportAdd] (/opt/opencti/build/src/resolvers/report.js:74:43)\n    at fieldResolver (/opt/opencti/build/src/graphql/authDirective.js:59:24)\n    at resolveFn (/opt/opencti/build/node_modules/apollo-server-core/src/utils/schemaInstrumentation.ts:106:18)\n    at executeField (/opt/opencti/build/node_modules/graphql/execution/execute.js:492:20)\n    at callbackFn (/opt/opencti/build/node_modules/graphql/execution/execute.js:377:22)\n    at promiseReduce (/opt/opencti/build/node_modules/graphql/jsutils/promiseReduce.js:23:9)\n    at executeFieldsSerially (/opt/opencti/build/node_modules/graphql/execution/execute.js:373:29)\n    at executeOperation (/opt/opencti/build/node_modules/graphql/execution/execute.js:347:14)\n    at execute (/opt/opencti/build/node_modules/graphql/execution/execute.js:136:20)\n    at execute (/opt/opencti/build/node_modules/apollo-server-core/src/requestPipeline.ts:504:20)\n    at processGraphQLRequest (/opt/opencti/build/node_modules/apollo-server-core/src/requestPipeline.ts:410:28)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at processHTTPRequest (/opt/opencti/build/node_modules/apollo-server-core/src/runHttpQuery.ts:437:24)"
    }
  ],
  "inner_relation_creation": 0,
  "level": "error",
  "message": "Platform unmanaged direct error",
  "operation": "ReportCreationMutation",
  "operation_query": "mutation ReportCreationMutation($input:ReportAddInput!){reportAdd(input:$input){id standard_id name description entity_type confidence parent_types ...ReportLine_node}}fragment ReportLine_node on Report{id entity_type name description published report_types createdBy{__typename __isIdentity:__typename id name entity_type}objectMarking{id definition_type definition x_opencti_order x_opencti_color}objectLabel{id value color}creators{id name}status{id order template{name color id}}workflowEnabled created_at}",
  "size": 448,
  "time": 8,
  "timestamp": "2024-02-22T09:35:09.680Z",
  "type": "WRITE_ERROR",
  "user": {
    "group_ids": [
      "a073e696-f9b8-4846-afa2-02c55fb33854",
      "dd020492-f39e-411b-b354-474b1b20ce7e"
    ],
    "ip": "93.42.69.98",
    "organization_ids": [],
    "referer": "
[https://testing.octi.staging.filigran.io/dashboard/analyses/reports"](https://testing.octi.staging.filigran.io/dashboard/analyses/reports%22)
,
    "socket": "query",
    "user_id": "a93d949b-b56d-4426-b7fe-b79ec3718b0e",
    "user_metadata": {}
  },
  "variables": {
    "input": {
      "confidence": 100,
      "content": "<p>This is a default content for all my reports that I will create 123</p>",
      "description": "",
      "externalReferences": [],
      "file": {
        "file": {
          "encoding": "7bit",
          "filename": "Capture d'écran 2023-04-17 133954.png",
          "mimetype": "image/png"
        },
        "promise": {}
      },
      "name": "Test report with file",
      "objectAssignee": [],
      "objectLabel": [],
      "objectMarking": [],
      "objectParticipant": [],
      "published": "2024-02-22T10:34:50+01:00",
      "report_types": []
    }
  },
  "version": "5.12.32"
}
```

### PR DESCRIPTION
**Pull request by Kedae**
Closes #6075

### CODE REVIEW

File: opencti-platform/opencti-front/tests_e2e/report/createReport.spec.ts [hunk 1]
```diff
@@ -14,3 +15,19 @@ test('Create a new report page', async ({ page }) => {
...
+  await page.goto('/dashboard/analyses/reports');
+  await reportPage.addNewReport();
+  await reportForm.fillNameInput('Test e2e with file');
+  const fileChooserPromise = page.waitForEvent('filechooser');
+  await page.getByRole('button', { name: 'Select your file', exact: true }).click();
+  const fileChooser = await fileChooserPromise;
+  await fileChooser.setFiles(path.join(__dirname, 'createReport.spec.ts'));
+  await reportPage.getCreateReportButton().click();
+  await reportPage.getItemFromList('Test e2e with file').click();
+  await expect(reportDetailsPage.getReportDetailsPage()).toBeVisible();
```
Comment (labo-flg) [inline opencti-platform/opencti-front/tests_e2e/report/createReport.spec.ts:32]: maybe add a test on the data tab > presence of the associated file in the list

File: opencti-platform/opencti-graphql/src/domain/report.js [hunk 2]
```diff
@@ -118,7 +118,7 @@ export const reportsDistributionByEntity = async (context, user, args) => {
 
 // region mutations
 export const addReport = async (context, user, report) => {
-  const finalReport = R.assoc('created', report.published, report);
+  const finalReport =  { ...R.assoc('created', report.published, report), file: await report.file };
```
Comment (labo-flg) [inline opencti-platform/opencti-graphql/src/domain/report.js]: so there was another issue actually?

Comment (Kedae) [reply opencti-platform/opencti-graphql/src/domain/report.js]: Nop, forgot to revert that try :D

File: opencti-platform/opencti-front/tests_e2e/report/createReport.spec.ts (hunk 1 above)
Comment (Kedae) [reply opencti-platform/opencti-front/tests_e2e/report/createReport.spec.ts:32]: Done

**Review by SouadHadjiat — APPROVED** [review]

File: opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ test('Check EE activation', async ({ page }) => {
   await page.getByRole('button', { name: 'Enable' }).click();
   expect(page.getByText(/^Enterprise$/)).toBeVisible();
   await page.getByRole('button', { name: 'Disable Enterprise Edition' }).click();
-  expect(page.getByText(/^Enterprise$/)).toBeVisible();
+  expect(page.getByText(/^Community$/)).toBeVisible();
```
Comment (labo-flg) [inline opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts]: unrelated to this PR but that triggers a legit question on making e2e tests with or without entreprise edition enabled

**Review by labo-flg — APPROVED** [review]

File: opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts (hunk 3 above)
Comment (labo-flg) [reply opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts]: and this seems to fail in drone actually

**Pull request merged**